src/python/pants/base/fingerprint_strategy.py

```diff
@@ -39,6 +39,8 @@ def compute_fingerprint(self, target):
   def fingerprint_target(self, target):
     """Consumers of subclass instances call this to get a fingerprint labeled with the name"""
     fingerprint = self.compute_fingerprint(target)
+    # TODO: We probably don't need to qualify the fingerprint with the strategy name any more,
```

Comment:
If you're going to have multiple invalidated blocks, you could key each block in a different way?